### PR TITLE
[fix bug 1350170] Add Firefox Focus page

### PR DESCRIPTION
**Pull request by alexgibson**
## Description
- Adds Firefox focus product page at `/firefox/focus/`
- Still waiting on final copy approval, so do-not-merge for now.

## Bugzilla link
https://bugzilla.mozilla.org/show_bug.cgi?id=1350170

## Testing
Demo: https://bedrock-demo-agibson.us-west.moz.works/en-US/firefox/focus/

## Checklist
- [ ] Requires l10n changes.
- [ ] Related functional & integration tests passing.

### CODE REVIEW

**Comment by alexgibson** [comment]
This is still waiting on final copy, but should be good for review.

File: bedrock/settings/appstores.py [hunk 1]
```diff
@@ -85,3 +85,16 @@
...
+# Link to Firefox Focus for iOS on the Apple App Store via app.adjust.
+# ?fallback parameter sends users who are on the wrong platform to the iTunes website.
+# https://bugzilla.mozilla.org/show_bug.cgi?id=1350170
+APPLE_APPSTORE_FIREFOX_FOCUS_LINK = ('https://app.adjust.com/b8s7qo?campaign=&adgroup=&creative=iOS' +
+                                     '&fallback=https%3A%2F%2Fitunes.apple.com%2Fus%2Fapp%2Fid1055677337%3Fmt%3D8')
+
+# Link to Firefox Focus for Android on the Google Play Store via app.adjust.
+# ?fallback parameter sends users who are on the wrong platform to the Play Store website.
+# https://bugzilla.mozilla.org/show_bug.cgi?id=1350170
+# TODO - this is still the wrong URL
```
Comment (alexgibson) [inline bedrock/settings/appstores.py]: TODO for myself, update this link

**Comment by alexgibson** [comment]
This one is due to go out on Tuesday, so having it reviewed before then would be the ideal.

File: bedrock/settings/appstores.py [hunk 2]
```diff
@@ -85,3 +85,16 @@
...
+# Link to Firefox Focus for iOS on the Apple App Store via app.adjust.
+# Fallback parameter sends users who are on the wrong platform to the iTunes website.
+# https://bugzilla.mozilla.org/show_bug.cgi?id=1350170
+APPLE_APPSTORE_FIREFOX_FOCUS_LINK = ('https://app.adjust.com/b8s7qo?campaign=&adgroup=Focus&creative=iOS&' +
+                                     'fallback=https%3A%2F%2Fitunes.apple.com%2Fapp%2Fid1055677337%3Fmt%3D8')
+
+# Link to Firefox Focus for Android on the Google Play Store via app.adjust.
+# Fallback parameter sends users who are on the wrong platform to the Play Store website.
+# https://bugzilla.mozilla.org/show_bug.cgi?id=1350170
+# TODO - this is still the wrong fallback URL, waiting on app being published.
```
Comment (alexgibson) [inline bedrock/settings/appstores.py]: Still need to update this link on Tuesday

File: bedrock/firefox/templates/firefox/products/focus.html [hunk 3]
```diff
@@ -0,0 +1,154 @@
+{# This Source Code Form is subject to the terms of the Mozilla Public
+ # License, v. 2.0. If a copy of the MPL was not distributed with this
+ # file, You can obtain one at http://mozilla.org/MPL/2.0/. #}
+
+{% extends "firefox/products/product-base.html" %}
+
+{# L10n: strings for this page are not considered final and aren't ready for localization. #}
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/products/focus.html]: Note: I'm going to update this in #4908 once this PR merges. We may need to add some helpful l10n notes or alternative strings for the headings below.

File: bedrock/settings/appstores.py [hunk 4]
```diff
@@ -85,3 +85,16 @@
     'vi': 'vt',     # Vietnam
     'en-ZA': 'za',  # South Africa
 }
+
+# Link to Firefox Focus for iOS on the Apple App Store via app.adjust.
+# Fallback parameter sends users who are on the wrong platform to the iTunes website.
+# https://bugzilla.mozilla.org/show_bug.cgi?id=1350170
+APPLE_APPSTORE_FIREFOX_FOCUS_LINK = ('https://app.adjust.com/b8s7qo?campaign=&adgroup=Focus&creative=iOS&' +
```
Comment (jpetto) [inline bedrock/settings/appstores.py]: Is the `campaign` param supposed to be empty here? It's set to `moz.org` in the Android link below...

File: tests/functional/firefox/test_focus.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+def test_app_store_button_is_displayed(base_url, selenium):
+    page = FirefoxFocusPage(selenium, base_url).open()
+    assert page.is_app_store_button_displayed
+    assert page.is_play_store_button_displayed
+
+
+@pytest.mark.nondestructive
+def test_news_feed_is_displayed(base_url, selenium):
+    page = FirefoxFocusPage(selenium, base_url).open()
+    assert len(page.news_feed.articles) == 3
```
Comment (jpetto) [inline tests/functional/firefox/test_focus.py]: We should now drop these news feed tests/properties, right?

File: bedrock/settings/appstores.py (hunk 4 above)
Comment (alexgibson) [inline bedrock/settings/appstores.py]: Nice catch - I think you're right. I'm confirming with mhan in the bug, but getting two fully formed URLs pasted there is currently the biggest challenge for this page it seems 😩.

File: tests/functional/firefox/test_focus.py (hunk 5 above)
Comment (alexgibson) [inline tests/functional/firefox/test_focus.py]: Ah yes, will remove 👍

**Comment by alexgibson** [comment]
> The nav, however, is not looking very good.

Yup, we all knew the nav wasn't going to work well as soon as it was designed. It is however currently readable and accessible, and no longer blocks l10n. I think that's as much as we can do until creative / UX have time to work on this.

**Comment by alexgibson** [comment]
Thanks for the review @jpetto - just waiting on the final play store link and then I think we'll be good to go here 👍

File: bedrock/firefox/templates/firefox/products/focus.html [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+
+      <section id="modern" class="card card-modern">
+        <div class="card-content">
+          <a href="{{ url('firefox.features.fast') }}" data-link-type="link" data-link-position="Value - Modern" data-link-name="More Performance">
+            <h2 class="card-title">{{ _('More Performance') }}</h2>
+          </a>
+          <p>
+          {% trans %}
+            Bring it, Internet! We’ve spent the last year supercharging Firefox’s
+            performance. Now start up to 30% faster, tab hop 86% quicker, and
```
Comment (flodolo) [inline bedrock/firefox/templates/firefox/products/focus.html]: tab or tabs?

Comment (alexgibson) [inline bedrock/firefox/templates/firefox/products/focus.html]: This string is changing in #4908, please review l10n there and not here.

**Comment by alexgibson** [comment]
@jpetto - OK, updated URLs with the android play store link. Fabio says we have the green light to push this page out.

**Pull request merged**